File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:

        my_str = ""

        for i in s:
            if i.isalnum():
                my_str = my_str + i

        my_str = my_str.lower()
        reversed_my_str = my_str[::-1]

        if reversed_my_str == my_str:
            return True
        else:
            return False

File: 0125-valid-palindrome/0125-valid-palindrome.c
bool isPalindrome(char* s) {
    
    int i = 0;
    int len = strlen(s);
    for (i; i < len; i++){
      s[i] = tolower(s[i]);
    }
    
    int left_pointer = 0;
    int right_pointer = len-1;
    
    while(left_pointer < right_pointer){
      
      while(!isalnum(s[left_pointer]) && (left_pointer < right_pointer) ){
        left_pointer++;
      }
      while( !isalnum(s[right_pointer]) && right_pointer > left_pointer){
        right_pointer--;
      }
      
      if(s[left_pointer] != s[right_pointer]){
        return false;
      }
      
      left_pointer++;
      right_pointer--;
      
    }
    return true;
}